dd-sdk-android/src/main/kotlin/com/datadog/android/tracing/TracingInterceptor.kt

```diff
@@ -287,7 +294,9 @@ open class TracingInterceptor
             (span as? MutableSpan)?.setResourceName(RESOURCE_NAME_404)
         }
         onRequestIntercepted(request, span, response, null)
-        span?.finish()
+        if (canSendSpan()) {
```

Comment:
have you checked what happens into the DDTracer if the span is not finished to see if it is not trailing around ? In that case maybe we should add an internal method to clean that span ?

Reply:
Yes, the spans are not kept by the Tracer, and since we don't create a scope out of it, there's no reference outside of the one we keep in the Interceptor (and we only keep it for the duration of the request)